### DISCUSSION TITLE
Users cannot access web pages after connecting to VPN

### DISCUSSION THREAD

**Discussion by aabababb** [strongswan/strongswan#2124] [Q&A] [answered]
Users cannot access web pages after connecting to VPN

centos7 strongswan 5.9.13

log

Feb 27 15:16:07 test3 ipsec: 12[KNL] querying policy 0.0.0.0/0 === 192.168.88.1/32 out
Feb 27 15:16:07 test3 ipsec: 12[IKE] sending keep alive to 1.2.3.4[38548]
Feb 27 15:16:07 test3 ipsec: 11[NET] received packet: from 1.2.3.4[38548] to 10.0.0.60[4500] (68 bytes)
Feb 27 15:16:07 test3 ipsec: 11[ENC] parsed INFORMATIONAL request 6 [ D ]
Feb 27 15:16:07 test3 ipsec: 11[IKE] received DELETE for IKE_SA IKEv2[2]
Feb 27 15:16:07 test3 ipsec: 11[IKE] deleting IKE_SA IKEv2[2] between 10.0.0.60[test.net]...1.2.3.4[192.168.30.101]
Feb 27 15:16:07 test3 ipsec: 11[IKE] IKE_SA IKEv2[2] state change: ESTABLISHED => DELETING
Feb 27 15:16:07 test3 ipsec: 11[IKE] IKE_SA deleted
Feb 27 15:16:07 test3 ipsec: 11[ENC] generating INFORMATIONAL response 6 [ ]
Feb 27 15:16:07 test3 ipsec: 11[NET] sending packet: from 10.0.0.60[4500] to 1.2.3.4[38548] (60 bytes)
Feb 27 15:16:07 test3 ipsec: 11[IKE] IKE_SA IKEv2[2] state change: DELETING => DESTROYING
Feb 27 15:16:07 test3 ipsec: 11[KNL] deleting policy 0.0.0.0/0 === 192.168.88.1/32 out
Feb 27 15:16:07 test3 ipsec: 11[KNL] deleting policy 192.168.88.1/32 === 0.0.0.0/0 in
Feb 27 15:16:07 test3 ipsec: 11[KNL] deleting policy 192.168.88.1/32 === 0.0.0.0/0 fwd
Feb 27 15:16:07 test3 ipsec: 11[KNL] deleting SAD entry with SPI c370e57d
Feb 27 15:16:07 test3 ipsec: 11[KNL] deleted SAD entry with SPI c370e57d
Feb 27 15:16:07 test3 ipsec: 11[KNL] deleting SAD entry with SPI 015e4f05
Feb 27 15:16:07 test3 ipsec: 11[KNL] deleted SAD entry with SPI 015e4f05
Feb 27 15:16:07 test3 ipsec: 11[CFG] lease 192.168.88.1 by 'test' went offline

Feb 27 15:23:54 10[MGR] <IKEv2|5> checkin of IKE_SA successful
Feb 27 15:23:54 01[JOB] next event in 2s 479ms, waiting
Feb 27 15:23:57 01[JOB] got event, queuing job for execution
Feb 27 15:23:57 01[JOB] next event in 26s 519ms, waiting
Feb 27 15:23:57 09[MGR] checkout IKEv2 SA with SPIs 065d06_i a993c8910_r
Feb 27 15:23:57 09[MGR] IKE_SA IKEv2[5] successfully checked out
Feb 27 15:23:57 09[KNL] <IKEv2|5> querying policy 0.0.0.0/0 === 192.168.88.1/32 out
Feb 27 15:23:57 09[KNL] <IKEv2|5> sending XFRM_MSG_GETPOLICY 245: => 80 bytes @ 0x7f6ddc
Feb 27 15:23:57 09[KNL] <IKEv2|5>    0: 50 00 00 00 15 00 01 00 F5 00 00 00 AD 22 00 00  P............"..

Feb 27 15:23:57 09[KNL] <IKEv2|5> received XFRM_MSG_NEWPOLICY 245: => 264 bytes @ 0x7f6dc0
Feb 27 15:23:57 09[KNL] <IKEv2|5>    0: 08 01 00 00 13 00 00 00 F5 00 00 00 AD 22 00 00  ............."..

Feb 27 15:23:57 09[IKE] <IKEv2|5> sending keep alive to 1.2.3.4[38557]
Feb 27 15:23:57 09[MGR] <IKEv2|5> checkin IKEv2 SA IKEv2[5] with SPIs 065d063_i a993c8_r
Feb 27 15:23:57 09[MGR] <IKEv2|5> checkin of IKE_SA successful
Feb 27 15:23:57 04[NET] sending packet: from 10.0.0.60[4500] to 1.2.3.4[38557]
Feb 27 15:23:57 01[JOB] next event in 19s 999ms, waiting

any help?

**Comment by tobiasbrunner** [comment]
Not sure what you are trying to point out with your log. It shows an SA getting deleted by a client and a keep alive getting sent for some other SA.

Anyway, regarding forwarding client traffic, you might want to refer to [the docs](https://docs.strongswan.org/docs/5.9/howtos/forwarding.html).

**Reply by aabababb** [reply]
I set iptables,still not work.

iptables -t nat -A POSTROUTING -s 10.0.0.0/24,192.168.88.0/24 -o ens3 -m policy --dir out --pol ipsec -j ACCEPT
iptables -t nat -A POSTROUTING -s 10.0.0.0/24,192.168.88.0/24 -o ens3 -j MASQUERADE

iptables -t nat -I POSTROUTING -m policy --pol ipsec --dir out -j ACCEPT


iptables -t mangle -A FORWARD -m policy --pol ipsec --dir in  -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
iptables -t mangle -A FORWARD -m policy --pol ipsec --dir out -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360

how can I find the error log about this connecting?

**Reply by tobiasbrunner** [reply]
> I set iptables,still not work.

Without seeing what other firewall rules there are, we can't really help you. Please see #196.

> how can I find the error log about this connecting?

Not sure what error log you expect there to be about this, IPsec traffic (or network traffic in general) is not logged anywhere. Capturing traffic with tcpdump/Wireshark might help to see what happens to it, or tracing packets through Netfilter.

**Reply by aabababb** [reply]
# Generated by iptables-save v1.4.21 on Tue Feb 27 23:25:54 2024
*nat
:PREROUTING ACCEPT [55:4040]
:INPUT ACCEPT [41:3400]
:OUTPUT ACCEPT [127:7908]
:POSTROUTING ACCEPT [0:0]
-A POSTROUTING -m policy --dir out --pol ipsec -j ACCEPT
-A POSTROUTING -j MASQUERADE
-A POSTROUTING -s 192.168.88.0/24 -o ens3 -j MASQUERADE
-A POSTROUTING -s 10.0.0.0/24 -o ens3 -m policy --dir out --pol ipsec -j ACCEPT
-A POSTROUTING -s 192.168.88.0/24 -o ens3 -m policy --dir out --pol ipsec -j ACCEPT
-A POSTROUTING -s 10.0.0.0/24 -o ens3 -j MASQUERADE
COMMIT
# Completed on Tue Feb 27 23:25:54 2024
# Generated by iptables-save v1.4.21 on Tue Feb 27 23:25:54 2024
*filter
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [1400479:327647784]
-A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT
-A INPUT -p icmp -j ACCEPT
-A INPUT -i lo -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 22 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 500 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 25 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 110 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 80 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 139 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 443 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 445 -j ACCEPT
-A INPUT -p udp -m policy --dir in --pol ipsec -m udp --dport 1701 -j ACCEPT
-A INPUT -p udp -m udp --dport 31701 -j ACCEPT
-A INPUT -p udp -m udp --dport 500 -j ACCEPT
-A INPUT -p udp -m udp --dport 4500 -j ACCEPT
-A INPUT -p esp -j ACCEPT
-A INPUT -m policy --dir in --pol ipsec -j ACCEPT
-A INPUT -j REJECT --reject-with icmp-host-prohibited
-A INPUT -p tcp -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A INPUT -p udp -m udp --dport 53 -m conntrack --ctstate NEW -j ACCEPT
-A INPUT -p tcp -m tcp --dport 22 -m conntrack --ctstate NEW -j ACCEPT
-A INPUT -p tcp -m tcp --dport 1723 -m conntrack --ctstate NEW -j ACCEPT
-A INPUT -p tcp -m tcp --dport 47 -m conntrack --ctstate NEW -j ACCEPT
-A INPUT -p tcp -m tcp --dport 1701 -m conntrack --ctstate NEW -j ACCEPT
-A INPUT -p udp -m multiport --dports 500,4500,1701 -j ACCEPT
-A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -s 10.0.0.0/24 -i ens3 -j ACCEPT
-A FORWARD -d 10.0.0.0/24 -o ens3 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -s 10.0.0.0/24 -o ens3 -j ACCEPT
-A FORWARD -s 10.0.0.0/24 -i ens3 -p udp -m udp --dport 4500 -j ACCEPT
-A FORWARD -d 10.0.0.0/24 -o ens3 -p udp -m udp --sport 4500 -j ACCEPT
-A FORWARD -s 10.0.0.0/24 -i ens3 -p udp -m udp --dport 500 -j ACCEPT
-A FORWARD -d 10.0.0.0/24 -o ens3 -p udp -m udp --sport 500 -j ACCEPT
-A FORWARD -j DROP
-A FORWARD -j REJECT --reject-with icmp-host-prohibited
-A FORWARD -i ppp+ -m state --state NEW,RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -m state --state RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -s 192.168.88.0/24 -j ACCEPT
COMMIT
# Completed on Tue Feb 27 23:25:54 2024




all the  iptables rules,thank you take a look.

**Reply by tobiasbrunner** [reply]
There seem to be a lot of redundant rules there (e.g. DROP and REJECT for FORWARD or duplicate NAT rules), and some useless ones as well (FORWARD rules for IKE traffic from virtual IPs?, or accepting traffic to UDP port 500 twice, or rules following unqualified DROP/REJECT rules). Regarding the FORWARD rules, you have ACCEPT rules for `10.0.0.0/24`, however, the virtual IPs, according to the log, are in `192.168.88.0/24`. There is an ACCEPT rule for the latter, but it comes after the DROP and REJECT rules, so this won't have any effect. You might want to start from scratch with your firewall config.

**Reply by aabababb** [reply]
```
# Generated by iptables-save v1.4.21 on Thu Feb 29 15:26:18 2024
*mangle
:PREROUTING ACCEPT [8351:4045050]
:INPUT ACCEPT [8351:4045050]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [10452:2403987]
:POSTROUTING ACCEPT [10452:2403987]
-A FORWARD -m policy --pol ipsec --dir in  -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
-A FORWARD -m policy --pol ipsec --dir out -p tcp -m tcp --tcp-flags SYN,RST SYN -m tcpmss --mss 1361:1536 -j TCPMSS --set-mss 1360
COMMIT
# Completed on Thu Feb 29 15:26:18 2024
# Generated by iptables-save v1.4.21 on Thu Feb 29 15:26:18 2024
*nat
:PREROUTING ACCEPT [4:168]
:INPUT ACCEPT [0:0]
:OUTPUT ACCEPT [11:664]
:POSTROUTING ACCEPT [11:664]
-A POSTROUTING -s 192.168.88.0/24 -o ens3 -m policy --dir out --pol ipsec -j ACCEPT
-A POSTROUTING -s 192.168.88.0/24 -o ens3 -j MASQUERADE
-I POSTROUTING -m policy --pol ipsec --dir out -j ACCEPT
COMMIT
# Completed on Thu Feb 29 15:26:18 2024
# Generated by iptables-save v1.4.21 on Thu Feb 29 15:26:18 2024
*filter
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [379:79472]
-A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT
-A INPUT -p icmp -j ACCEPT
-A INPUT -i lo -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 22 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 21 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 53 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 25 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 110 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 80 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 139 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 443 -j ACCEPT
-A INPUT -p tcp -m state --state NEW -m tcp --dport 445 -j ACCEPT
-A INPUT -p udp -m udp --dport 53 -m conntrack --ctstate NEW -j ACCEPT
-A INPUT -p tcp -m tcp --dport 22 -m conntrack --ctstate NEW -j ACCEPT
-A INPUT -p tcp -m tcp --dport 1723 -m conntrack --ctstate NEW -j ACCEPT
-A INPUT -p tcp -m tcp --dport 47 -m conntrack --ctstate NEW -j ACCEPT
-A INPUT -p udp -m udp --dport 500 -j ACCEPT
-A INPUT -p udp -m udp --dport 4500 -j ACCEPT
-A INPUT -j REJECT --reject-with icmp-host-prohibited
-A FORWARD -s 192.168.88.0/24 -j ACCEPT
-A FORWARD -j REJECT --reject-with icmp-host-prohibited
COMMIT
# Completed on Thu Feb 29 15:26:18 2024
```

I reset the iptables,is it right now? but it still unable to access the external network.

**Reply by tobiasbrunner** [reply, marked as answer]
> -A FORWARD -s 192.168.88.0/24 -j ACCEPT
> -A FORWARD -j REJECT --reject-with icmp-host-prohibited

Well, that FORWARD chain is obviously incomplete. Return traffic will not have a source address in that subnet, so that will get rejected. Note that you could use strongSwan's default [updown script](https://docs.strongswan.org/docs/5.9/plugins/updown.html) to install connection-specific rules in the FORWARD chain, that might be helpful if you don't understand how this works.

**Reply by aabababb** [reply]
thanks a lot
